chef/cookbooks/barclamp/libraries/nic.rb

```diff
@@ -751,6 +766,50 @@ def remove_slave(slave)
       ::Kernel.system("ovs-vsctl del-port #{@nic} #{slave}")
       slave.down
     end
+
+    def up
+      slaves.each(&:up)
+      super
+    end
+
+    def destroy
+      slaves.each do |slave|
+        remove_slave(slave)
+      end
+      super
+      ::Kernel.system("ovs-vsctl del-br #{@nic}")
+      nil
+    end
+
+    def replug(slave)
+      slave = self.class.coerce(slave)
+      unless self.slaves.member?(slave)
+        raise ::ArgumentError.new("#{slave} is not a member of bridge #{@nic}")
+      end
+      ::Kernel.system("ovs-vsctl del-port #{@nic} #{slave}")
+      ::Kernel.system("ovs-vsctl add-port #{@nic} #{slave}")
+    end
+
+    def self.create(nic, slaves = [])
+      Chef::Log.info("Creating new OVS bridge #{nic}")
+      if self.exists?(nic)
+        raise ::ArgumentError.new("#{nic} already exists.")
+      end
+      ::Kernel.system("ovs-vsctl add-br #{nic}")
+      5.times do
```

Comment:
I am not entirely sure (I just copied this from the Bridge class), but my assumption was that it is just waiting for the sysfs file for the bridge to show up. It's just the ```self.exists?(nic) && self.ovs_bridge?(nic)``` that is executed multiple times. The code would be racey otherwise. 

I'll add a comment.